tests/pfcwd/test_pfcwd_function.py

```diff
@@ -0,0 +1,518 @@
+        ignore_file = os.path.join(TEMPLATES_DIR, "ignore_pfc_wd_messages")
+        reg_exp = loganalyzer.parse_regexp_file(src=ignore_file)
+        loganalyzer.ignore_regex.extend(reg_exp)
+        loganalyzer.expect_regex = []
+        loganalyzer.expect_regex.extend([EXPECT_PFC_WD_RESTORE_RE])
+        loganalyzer.match_regex = []
+
+        self.storm_hndle.stop_storm()
+        time.sleep(self.timers['pfc_wd_wait_for_restore_time'])
+        # storm restore
+        logger.info("Verify if PFC storm is restored on port {}".format(port))
+        loganalyzer.analyze(marker)
+        self.stats.get_pkt_cnts(self.queue_oid, begin=False)
+
+    def run_test(self, dut, port, action):
+        """
+        Test method that invokes the storm detection and restoration path which includes the traffic
+        test and associated counter verifications
+
+        Args:
+            dut(AnsibleHost) : DUT instance
+            port(string) : DUT port
+            action(string) : PTF test action
+        """
+        logger.info("--- Storm detection path for port {} ---".format(port))
+        loganalyzer = self.storm_detect_path(dut, port, action)
+        logger.info("--- Storm restoration path for port {} ---".format(port))
+        self.storm_restore_path(loganalyzer, port, action)
+        logger.info("--- Verify PFCwd counters for port {} ---".format(port))
+        self.stats.verify_pkt_cnts(self.pfc_wd['port_type'], self.pfc_wd['test_pkt_count'])
+
+    def test_pfcwd_actions(self, request, setup_pfc_test, fanout_graph_facts, ptfhost, duthost, fanouthosts):
+        """
+        PFCwd functional test
+
+        Args:
+            request(object) : pytest request object
+            setup_pfc_test(fixture) : Module scoped autouse fixture for PFCwd
+            fanout_graph_facts(fixture) : fanout graph info
+            ptfhost(AnsibleHost) : ptf host instance
+            duthost(AnsibleHost) : DUT instance
+            fanouthosts(AnsibleHost): fanout instance
+        """
+        setup_info = setup_pfc_test
+        self.fanout_info = fanout_graph_facts
+        self.ptf = ptfhost
+        self.dut = duthost
+        self.fanout = fanouthosts
+        self.timers = setup_info['pfc_timers']
+        self.ports = setup_info['selected_test_ports']
+        self.neighbors = setup_info['neighbors']
+        dut_facts = self.dut.setup()['ansible_facts']
+        self.peer_dev_list = dict()
+
+        for idx, port in enumerate(self.ports):
+             logger.info("")
+             logger.info("--- Testing various Pfcwd actions on {} ---".format(port))
+             self.setup_test_params(port, setup_info['vlan'], init=not idx)
+             self.traffic_inst = SendVerifyTraffic(self.ptf, dut_facts['ansible_eth0']['macaddress'], self.pfc_wd)
+             pfc_wd_restore_time_large = request.config.getoption("--restore-time")
+             self.timers['pfc_wd_wait_for_restore_time'] = int(pfc_wd_restore_time_large / 1000 * 2)
```

Comment:
Two q's: can we just do  `/ 2000`, and also where does this number come from?

Reply:
I believe you meant `/ 500`

Reply:
I guess the reason it was specified this way is to make it more explicit. converting the time to sec and then multiplying by 2. This is just a wait time to check the logs once the storm is restored. leaving it this way is easier for someone to understand. I will add in a comment

Reply:
That makes sense to me! It might be good to include the units in the comment so that it's clear where the `/1000` is coming from.